spec/saml_authenticator_spec.rb

```diff
@@ -34,6 +34,44 @@
 
       result = @authenticator.after_authenticate(hash)
       expect(result.user.email).to eq(@user.email)
+      expect(result.email_valid).to eq(true)
+    end
+
+    it 'defaults email_valid to false if saml_default_emails_valid is false' do
+      GlobalSetting.stubs(:saml_default_emails_valid).returns("false")
+
+      hash = OmniAuth::AuthHash.new(
+        uid: @uid,
+        info: {
+            name: "John Doe",
+            email: @user.email
+        }
+      )
+
+      result = @authenticator.after_authenticate(hash)
+      expect(result.user.email).to eq(@user.email)
+      expect(result.email_valid).to eq(false)
+    end
+
+    it 'defaults email_valid based on saml_validate_email_fields setting' do
+      GlobalSetting.stubs(:saml_validate_email_fields).returns("customers")
+
+      hash = OmniAuth::AuthHash.new(
+        uid: @uid,
+        info: {
+            name: "John Doe",
+            email: @user.email
+        },
+        extra: {
+          raw_info: OneLogin::RubySaml::Attributes.new({
+            'memberOf' => %w(Customers Employees)
```

Comment:
Should this not be case sensitive?

Reply:
The capitalization here is intentional to verify `(GlobalSetting.try(:saml_validate_email_fields).split("|").map(&:downcase) & auth.extra[:raw_info].attributes['memberOf'].map(&:downcase)).empty?` is working as expected. Notice `.map(&:downcase)`.

Reply:
@SamSaffron if you meant we should match case as well, I can do that. Was not sure about that.

Reply:
Mixed on this. I say leave it for now.